test/unit/deferred.js

```diff
@@ -525,6 +525,66 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
+		assert.ok( /barf/.test( msg ), "Message: " + msg );
+	};
+	jQuery.when(
+		defer.then( function() {
+			// Should get an error
+			jQuery.barf();
+		} ).then( null, jQuery.noop ),
+		defer.then( function() {
+			// Should NOT get an error
+			throw new Error( "Make me a sandwich" );
+		} ).then( null, jQuery.noop )
+	).then( function( ) {
+		window.console.warn = oldWarn;
+		done();
+	} );
+
+	defer.resolve();
+} );
+
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook with stack hooks", function( assert ) {
+
+	assert.expect( 2 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	jQuery.Deferred.getStackHook = function() {
+		// Default exceptionHook assumes the stack is in a form console.warn can log,
+		// but a custom getStackHook+exceptionHook pair could save a raw form and
+		// format it to a string only when an exception actually occurs.
+		// For the unit test we just ensure the plumbing works.
+console.trace();
```

Comment:
This needs to be removed but I wanted to be sure we had the valid stack at this point of execution.